InsightIDR: Multi Country Auth and Remediation (#62)
rapid7/insightconnect-workflows: workflows/InsightIDR_MultiCountry_Auth_and_Remediation/help.md

# Description

> This workflow is designed to respond to InsightIDR's Multi Country Authentication alert by confirming if the user authenticated from more than 2 locations. With a flux of VPN loggings,a way to automatically reduce false positives would be to close alerts that have 2 or less locations or automatically respond to alerts with more than 2 locations. I this workflow, InsightConnect is responding by disabling the user account associated with more than 2 locations. 

Reviewer: ```suggestion
This workflow is designed to respond to InsightIDR's Multi Country Authentication alert by confirming if the user authenticated from more than two locations. Due to an influx of VPN logins, the amount of false positive alerts is increasing. This workflow will automatically reduce false positives by closing alerts with two or less locations or by automatically responding to alerts with more than two locations. If a user account is shown to be associated with two or more locations InsightConnect will respond by disabling the user account.
```

# Key Features

* Parses authentication count using Pattern Match & Python

> * If locations are 2 or less, the InsightIDR investigation is automatically closed.

Reviewer: ```suggestion
* If locations are two or less, the InsightIDR investigation is automatically closed.
```

> * If there are more than 2 locations Active Directory is automatically querying for the associated user. If the user is found, they are disabled.

Reviewer: ```suggestion
* If locations are more than two, Active Directory will query for the associated user and disable the account if found.
```

# Requirements

> * InsightConnect Orchestrator Deployed

Reviewer: ```suggestion
* InsightConnect Orchestrator deployed
```

> * Active Directory Connection

Reviewer: ```suggestion
* Active Directory connection
```

> * Insight Platform Organization Key

Reviewer: ```suggestion
* [Insight Cloud Platform API key](https://insight.rapid7.com/platform#/apiKeyManagement)
```

# Documentation

## Setup

> Once the workflow has been imported, edit the workflow and setup or select your Active Director connection in the Query and Disable user steps. Also setp up your InsightIDR connection with an Organization Key for the Insight Platform.

Reviewer: ```suggestion
Once the workflow has been imported, edit the workflow and either configure or select an existing Active Directory connection in the Query and Disable User steps. Your InsightIDR connection will also need to be configured using your Insight Cloud Platform API key.
```
